### DISCUSSION TITLE
Git remote and local操作問題

### DISCUSSION THREAD

**Discussion by michael61112** [bol-edu/HLS-SOC-Discussions#142] [Q&A]
Dear 講師，

若我從remote抓一包code下來，可能是課程的作業repo，也可能是open source repo，
我在local修改過後並不打算push 回原本的remote，
我想存在自己的github remote，
該怎麼操作才能完整保留原本git log紀錄與我自己新加的commit一起push到我的github呢?

目前我將課程的repo抓下來修改後想透過自己的github link交作業，
在push操作時卡關，我的解法是把.git刪掉後重新commit一筆initial patch，
但我覺得這樣很可惜，歷史紀錄都不見了。

BTW，
若原本的remote更新了，我一樣可以用fetch+merge後push 到我自己的github嗎?

Best Regards,
Michael Kao
email: michael61112@yahoo.com.tw

**Comment by holyuming** [comment]
Open source repo你需要先fork一份到自己的帳號，然後你再clone你的帳號下的這個repo，接下來你就可以push commit到自己的這個remote repo同時保有原本的git log

**Reply by michael61112** [reply]
OK 感謝~!
